app/Http/Controllers/OrderController.php

```diff
@@ -176,4 +176,28 @@ private function getProductOrderForUser(User $user, Product $product): ?Order
 
         return $order;
     }
+
+    /**
+     * @return JsonResponse
+     */
+    public function getAllOrdersByDateAndUser(): JsonResponse
+    {
+        $orders = Order::query()
+            ->with([
+                'deliverer',
+                'user',
+                'product'
+                ]
+            )
+            ->orderBy('date','DESC')
+            ->get()
+            ->groupBy([function($order) {
+                return Carbon::parse($order->date)->format('Ymd');
+            },'paid_by'])
+            ->map(function ($value, $key){
```

Comment:
Ik denk dat Laravel een dedicated methode heeft om keys in te stellen in een Collection